Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest/Contest.tsx

```diff
@@ -98,34 +99,82 @@ const Contest = () => {
-                <div className={navigationContestClassName}>
-                    <ContestTasksNavigation />
+    const determineErrorMessage = useCallback(() => {
+        if (!validationError.contestIsFound) {
+            return `${contest?.name} - ${ContestValidationErrors.NotFound}`;
+        } if (!validationError.contestIsNotExpired) {
+            return `${contest?.name} - ${ContestValidationErrors.Expired}`;
+        } if (!validationError.isParticipantRegistered) {
+            return `${contest?.name} - ${ContestValidationErrors.NotRegistered}`;
+        } if (!validationError.contestCanBeCompeted || !validationError.contestCanBePracticed) {
+            return `${contest?.name} - ${ContestValidationErrors.NotEligible}`;
+        }
+
+        return null;
+    }, [ contest, validationError ]);
+
+    const renderErrorMessage = useCallback(() => (
+        <div className={styles.headingContest}>
+            <Heading
+              type={HeadingType.primary}
+              className={styles.contestHeading}
+            >
+                {determineErrorMessage()}
+            </Heading>
+        </div>
+    ), [ determineErrorMessage ]);
+
+    const renderContest = useCallback(
+        () => (
+            <>
+                <div className={styles.headingContest}>
+                    <Heading
+                      type={HeadingType.primary}
+                      className={styles.contestHeading}
+                    >
+                        {contest?.name}
+                    </Heading>
+                    <Heading type={HeadingType.secondary} className={secondaryHeadingClassName}>
+                        {renderTimeRemaining()}
+                        {renderScore()}
+                    </Heading>
                 </div>
-                <div className={submissionBoxClassName}>
-                    <SubmissionBox />
-                </div>
-                <div className={problemInfoClassName}>
-                    <ContestProblemDetails />
+
+                <div className={styles.contestWrapper}>
+                    <div className={navigationContestClassName}>
+                        <ContestTasksNavigation />
+                    </div>
+                    <div className={submissionBoxClassName}>
+                        <SubmissionBox />
+                    </div>
+                    <div className={problemInfoClassName}>
+                        <ContestProblemDetails />
+                    </div>
                 </div>
-            </div>
-        </>
+            </>
+        ),
+        [
+            contest,
+            navigationContestClassName,
+            problemInfoClassName,
+            renderScore,
+            renderTimeRemaining,
+            secondaryHeadingClassName,
+            submissionBoxClassName,
+        ],
+    );
+
+    const renderPage = useCallback(
+        () => !isNil(renderErrorMessage())
+            ? renderErrorMessage()
+            : renderContest(),
```

Comment:
Invert them. Always put the _normal condition_ first, i.e. `isNill(...)` is the normal condition.

Reply:
Done.